compiler/rustc_typeck/src/check/writeback.rs

```diff
@@ -732,14 +732,40 @@ impl<'cx, 'tcx> Resolver<'cx, 'tcx> {
     }
 }
 
+struct EraseEarlyRegions<'tcx> {
+    tcx: TyCtxt<'tcx>,
+}
```

Comment:
Isn't there one of these already somewhere in the codebase?

Reply:
I thought so too, but I couldn't find one.

Reply:
I must have been thinking of the following, which looks like the exact opposite 😅

https://github.com/rust-lang/rust/blob/be76bdf9058172dd22a67332b45c0ab93268b3b3/compiler/rustc_typeck/src/hir_wf_check.rs#L169-L188